go/store/prolly/message/serialize.go

```diff
@@ -58,6 +59,25 @@ func writeItemOffsets(b *fb.Builder, items [][]byte, sumSz int) fb.UOffsetT {
 	return b.EndVector(cnt)
 }
 
+// writeAddrOffests returns offsets into the values array that correspond to addr root
+// hashes that themselves have subtrees
+func writeValAddrOffsets(b *fb.Builder, items [][]byte, sumSz int, valDesc val.TupleDesc) fb.UOffsetT {
+	var cnt int
+	var off = sumSz
+	for i := len(items) - 1; i >= 0; i-- {
+		tup := val.Tuple(items[i])
+		off -= len(tup) // start of tuple
+		for j := range valDesc.Addrs {
+			// get index into value tuple pointing at address
+			o, _ := tup.GetOffset(j)
```

Comment:
you need to check if field `j` is NULL, otherwise we'll be reading a nonsense address from the next field

Reply:
you need to account for null blob fields here

Reply:
in a different way than i already did?

Reply:
sorry, I was confused by the `_`, LGTM